### ISSUE TITLE
typeset and declare treated differently

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#2262]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2155
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

declare x=$(somecmd foo)
typeset y=$(somecmd foo)

echo "$x, $y"



```

#### Here's what shellcheck currently says:
```
Line 3	SC2155: Declare and assign separately to avoid masking return values.
```
#### Here's what I wanted or expected to see:
```
Line 3	SC2155: Declare and assign separately to avoid masking return values.
Line 4	SC2155: Declare and assign separately to avoid masking return values.
```

Why is typeset treated differently than declare?

**Comment by TinCanTech** [comment]
Given what `man bash` has to say, I would chalk-this-up to a simple minor omission by `shellcheck`

I can't find any _significant_ difference between `declare` vs `typeset`

**Comment by koalaman** [comment]
This was indeed a simple omission. It now triggers as expected, and is also better at detecting when a local read-only variable is declared in ways other than `local` (such as with `typeset -r` inside a Ksh function declared with the `function` keyword).

Thanks!
